iocore/net/SSLNetVConnection.cc

```diff
@@ -820,6 +825,7 @@ void
 SSLNetVConnection::do_io_close(int lerrno)
 {
   if (this->ssl != nullptr && sslHandShakeComplete) {
+    callHooks(TS_EVENT_VCONN_CLOSE);
```

Comment:
What happens if we cannot get the logic on the continuation?  Should this be like the SERVERNAME callback?  Lockless and synchronous.  It would be very difficult to keep state around while rescheduling the the callback on shutdown.

Reply:
Yes. I think that trying to reschedule in this case is too fraught to contemplate.